frontend/packages/ceph-storage-plugin/src/components/block-pool/create-block-pool.tsx

```diff
@@ -0,0 +1,171 @@
+      const poolObj: StoragePoolKind = getPoolKindObj(state);
+
+      dispatch({ type: BlockPoolActionType.SET_INPROGRESS, payload: true });
+      k8sCreate(CephBlockPoolModel, poolObj)
+        .then(() => history.push(`${storageClusterListPage}/${state.poolName}`))
+        .finally(() => dispatch({ type: BlockPoolActionType.SET_INPROGRESS, payload: false }))
+        .catch((err) =>
+          dispatch({
+            type: BlockPoolActionType.SET_ERROR_MESSAGE,
+            payload: getErrorMessage(err.message) || 'Could not create block pool.',
+          }),
+        );
+    }
+  }, [state, state.isSubmitted, storageClusterListPage]);
+
+  if (cephCluster?.metadata.name === CEPH_EXTERNAL_CR_NAME) {
+    return (
+      <Modal
+        title={t('ceph-storage-plugin~Create Block Pool')}
+        titleIconVariant="warning"
+        isOpen
+        onClose={onClose}
+        variant="small"
+        isFullScreen={false}
+        actions={[
+          <Button key="confirm" variant="primary" onClick={onClose}>
+            {t('ceph-storage-plugin~Close')}
+          </Button>,
+        ]}
+      >
+        <strong>
+          {t(
+            'ceph-storage-plugin~Pool creation is not available for openshift container storage external mode.',
+          )}
+        </strong>
+      </Modal>
+    );
+  }
+
+  return (
+    <>
+      <div className="co-create-operand__header">
+        <div className="co-create-operand__header-buttons">
+          {memoizedCSV !== null && (
+            <BreadCrumbs
+              breadcrumbs={[
+                {
+                  name: memoizedCSV.spec?.displayName,
+                  path: url.replace('/~new', ''),
+                },
+                {
+                  name: t('ceph-storage-plugin~Create Block Pool'),
+                  path: url,
+                },
+              ]}
+            />
+          )}
+        </div>
+
+        <h1 className="co-create-operand__header-text">
+          {t('ceph-storage-plugin~Create Block Pool')}
+        </h1>
+        <p className="help-block">
+          {t(
+            'ceph-storage-plugin~Pools are logical groups of Ceph objects. Such objects live inside of Ceph, or rather they live inside RADOS.',
+          )}
+        </p>
+      </div>
+      <div className="ceph-block-pool ceph-block-pool__form">
+        {isLoaded && !loadError ? (
+          <>
+            <PoolBody cephCluster={cephCluster} state={state} dispatch={dispatch} />
+            <div className="ceph-block-pool__footer">
+              <PoolFooter
+                poolName={state.poolName}
+                poolStatus={state.poolStatus}
+                replicaSize={state.replicaSize}
+                inprogress={state.inprogress}
+                errorMessage={state.errorMessage}
+                dispatch={dispatch}
+                cancel={onClose}
+              />
+            </div>
+          </>
+        ) : (
+          <LoadingInline />
```

Comment:
It will show loading even if there is a loadError, we want to stop on loadError.
Perhaps `StatusBox` component, you can wrap your component inside.

Reply:
ack